### ISSUE TITLE
Would `expect(promise).toReject()` be possible?

### ISSUE THREAD

**Issue by Stuk** [montagejs/jasminum#3]
Mocha-as-promised has `promise.should.eventually...`. To match our verbiage perhaps that would be `expect(promise).eventually.toThrow()`?

**Comment by kriskowal** [comment]
Yeah, `toBeRejected`, `toBeFulfilled` would match the API well. It would entail supporting a queue of promises in the test runner, exposing a way to enqueue more as the test runs.

I do think this would be in scope for Jasminum’s core.

**Comment by kriskowal** [comment]
But upon sitting down to implement this feature, it occurred to me that `toBeFulfilled` and `toBeRejected` were not satisfactory. We really would need all combinations of to eventually be, to eventually equal, to eventually have, to eventually throw etc across the spectrum of existing matchers, and even extended matchers. Ergo, we really would need an `eventually` property that would memoize a snapshot of all matchers and decorate them with a wrapper.
